server/api/src/rsDataObjOpen.cpp

```diff
@@ -710,13 +907,32 @@ int rsDataObjOpen(
         return l1descInx;
     }
 
+    using irods::experimental::key_value_proxy;
+
+    key_value_proxy kvp{dataObjInp->condInput};
+
+    // If the client specified a leaf resource, then resolve the hierarchy and store it in the keyValPair_t.
+    // This instructs the iRODS server to create the replica at the specified resource if it does not exist.
+    if (kvp.contains(LEAF_RESOURCE_NAME_KW)) {
```

Comment:
need a unit-test for this.

Reply:
this will likely be an istream test.

Reply:
This keyword allows any client to bypass voting and specify where a replica should be created, if it does not exist. Normally, clients are only allowed to specify a root resource and the server handles the rest.

Should this keyword be accessible only on the server-side?

Reply:
replace "resolve" with "discover" in comment.

It has been decided that "LEAF_RESOURCE_NAME_KW" should be a server-side thing only.

Reply:
this cannot be made a server-side only keyword due to the iRODS system not being able to distinguish clients and servers. this keyword does not introduce a new situation either. it only provides a convenient way to generate the resource hierarchy.

marking as resolved.